query/iterator.gen.go.tmpl

```diff
@@ -409,14 +415,44 @@ func (h *{{$k.name}}SortedMergeHeap) Less(i, j int) bool {
 		} else if xTags, yTags := x.Tags.Subset(h.opt.Dimensions), y.Tags.Subset(h.opt.Dimensions); !xTags.Equals(&yTags) {
 			return xTags.ID() < yTags.ID()
 		}
+
+		if x.Time == y.Time {
+			if len(x.Aux) > 0 && len(x.Aux) == len(y.Aux) {
+				for i := 0; i < len(x.Aux); i++ {
+					v1, ok1 := x.Aux[i].(string)
+					v2, ok2 := y.Aux[i].(string)
+					if !ok1 || !ok2 {
+						// Unsupported types used in Aux fields. Maybe they
+						// need to be added here?
+						break
+					}
+					return v1 < v2
+				}
+			}
+		}
 		return x.Time < y.Time
 	}
```

Comment:
I think this function would flow better if you handle the time comparison first & then compare aux values, and then handle aux lengths last:

```go
if x.Time != y.Time {
	return x.Time < y.Time
}

for i := 0; i < len(x.Aux) && i < len(y.Aux); i++ {
	if cmp := Compare(x.Aux[i], y.Aux[i]); cmp != 0 {
		return cmp == -1
	}
}

return len(x.Aux) < len(y.Aux)
```

Then move the comparison function out:

```go
func Compare(a, b interface{}) int {
	switch a := a.(type) {
	case string:
		if b, ok := b.(string); ok {
			return strings.Compare(a, b)
		}
	}
	return 0
}
```